lib/bap_primus/bap_primus_lisp_program.ml

```diff
@@ -150,8 +150,8 @@ let pp_term pp_exp ppf = function
 let pp_word = pp_term Int64.pp
 let pp_var = pp_term String.pp
 
-let rec concat_prog =
-  List.concat_map ~f:(function
+let rec concat_prog p =
```

Comment:
yet another definition rejected by https://github.com/ocaml/ocaml/pull/556 PR.

Reply:
Are you sure you don't mean the opposite? OCaml 4.06 _requires_ eta expansion here--you ought to get an error without, explicitly rejecting the LHS without.

Reply:
Yeah, the new let-rec checker rejects this, and although I'm fine with it, I've submitted an issue and notified upstream, as I have an impression from the PR comments that is not expected. But in any case, eta-expansion here is fine.